modules/core/src/main/java/org/apache/ignite/internal/processors/cache/distributed/dht/preloader/GridDhtPartitionDemander.java

```diff
@@ -1641,17 +1723,41 @@ private synchronized void partitionDone(UUID nodeId, int p, boolean own) {
             assert rmvd : "Partition already done [grp=" + grp.cacheOrGroupName() + ", fromNode=" + nodeId +
                 ", part=" + p + ", left=" + parts + "]";
 
-                if (rmvd)
-                    partitionsLeft.decrementAndGet();
+            if (rmvd)
+                partitionsLeft.decrementAndGet();
 
             if (parts.isEmpty()) {
                 int remainingRoutines = remaining.size() - 1;
 
-                U.log(log, "Completed " + ((remainingRoutines == 0 ? "(final) " : "") +
+                int fullParts = fullRebalancingParts.get(nodeId).size();
+                int histParts = histRebalancingParts.get(nodeId).size();
```

Comment:
Let's get rid of fullRebalancingParts and histRebalancingParts. For example, you can get a number of full partitions for a node using information provided in `fut.rebalancingParts` and `fut.historical` using the filter like:
`rebalancingParts.get(null).stream().filter(p -> !historical.contains(p)).count();`